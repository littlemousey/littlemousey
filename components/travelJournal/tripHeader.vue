<template>
  <div>
    <v-row align="start" justify="center">
      <v-col cols="12">
        <v-card class="my-5">
          <v-img class="white--text" :src="imgUrl" height="300">
            <v-card-title class="align-end fill-height">{{
              title
            }}</v-card-title>
          </v-img>
        </v-card>
      </v-col>
    </v-row>
    <v-row justify="center"
      ><h1 class="display-3 text-center">{{ title }}</h1></v-row
    >
    <v-row class="my-10" justify="center">
      <v-col cols="10" md="6"
        ><div class="text-center">{{ text }}</div></v-col
      >
    </v-row>
  </div>
</template>

<script>
export default {
  name: 'TripHeader',
  props: {
    imgUrl: { type: String, required: true },
    title: { type: String, required: true },
    text: { type: String, required: true }
  }
}
</script>

<style lang="scss" scoped></style>
