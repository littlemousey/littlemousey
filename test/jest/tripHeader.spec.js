import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import tripHeader from '@/components/travelJournal/tripHeader'

Vue.use(Vuetify)

let wrapper
beforeEach(() => {
  wrapper = shallowMount(tripHeader, {
    propsData: {
      imgUrl: 'img.png',
      title: 'title',
      text: 'text'
    }
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Trip header', () => {
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
