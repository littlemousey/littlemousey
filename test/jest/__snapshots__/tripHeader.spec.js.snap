// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip header renders correctly 1`] = `
<div>
  <v-row-stub
    align="start"
    justify="center"
    tag="div"
  >
    <v-col-stub
      cols="12"
      tag="div"
    >
      <v-card-stub
        class="my-5"
        loaderheight="4"
        tag="div"
      >
        <v-img-stub
          class="white--text"
          height="300"
          position="center center"
          src="img.png"
          transition="fade-transition"
        >
          <v-card-title-stub
            class="align-end fill-height"
          >
            title
          </v-card-title-stub>
        </v-img-stub>
      </v-card-stub>
    </v-col-stub>
  </v-row-stub>
   
  <v-row-stub
    justify="center"
    tag="div"
  >
    <h1
      class="display-3 text-center"
    >
      title
    </h1>
  </v-row-stub>
   
  <v-row-stub
    class="my-10"
    justify="center"
    tag="div"
  >
    <v-col-stub
      cols="10"
      md="6"
      tag="div"
    >
      <div
        class="text-center"
      >
        text
      </div>
    </v-col-stub>
  </v-row-stub>
</div>
`;
